### DISCUSSION TITLE
OVPN - Building out this VPN network

### DISCUSSION THREAD

**Discussion by derekcentrico** [haugene/vpn-configs-contrib#111] [General]
I just switched to OVPN from NordVPN due to port forwarding issues.  At any rate, I'm only connecting to Swedish servers.  I see that the default.ovpn is linked to Sweden.ovpn within the repo.

1.  How can that be altered to go to a different country's OVPN?
2.  How best do I go about adding the remaining 17 countries to the project with them actually being utilized?

Happy for a couple week beer project to port in the OVPN configurations.  But, I don't want to spend the time if it won't be utilized because we default solely to Sweden for OVPN.

**Comment by derekcentrico** [comment]
Okay so, I've come a bit of a ways - https://github.com/derekcentrico/vpn-configs-contrib-ovpnwork/tree/main/openvpn/ovpn.  I need to screw around a bit more with updating/cleaning up the scripts.

My ignorance here is trying to use NordVPN as an example.  I have no training to lean on here.  I'm wondering about having a country selector, UDP vs. TCP selector, and standard vs. multi-hop selector.

My thoughts are that somehow a script could read the country abbreviations (first part of the ovpn name).  Next, I can have udp or tcp within the names.  Finally, I have multihop in the name if it's such a setup.

If anyone can help me that would be great.  At this point my copy/paste skills allowed me to mimic IPVanish by adding the random selecting bash script.

**Comment by derekcentrico** [comment]
Okay if I am reading https://github.com/haugene/docker-transmission-openvpn/blob/master/openvpn/start.sh correctly, then the main docker should see the "configure-openvpn.sh" script in OVPN.  So at least there's that - random selection can occur.  Not optimal by any means so figuring out how to select based on the filename would be great.  Now I'm a bit stumped.  (beer)

**Comment by derekcentrico** [comment]
I think I have stuff good to go.  My problem now is that it keeps trying to read the nonexistent default.ovpn instead of using the script.

https://github.com/derekcentrico/vpn-configs-contrib-ovpnwork/tree/main/openvpn/ovpn

@pkishino Mind helping me understand how can I get the docker to read the [configure-openvpn.sh](https://github.com/derekcentrico/vpn-configs-contrib-ovpnwork/blob/main/openvpn/ovpn/configure-openvpn.sh) file?

I added three new environmental vars to the docker-compose:
-       - OVPN_COUNTRY='us'
-       - OVPN_PROTOCOL='udp'
-       - OVPN_CONNECTION= 'standard'

These should provide the answers for the configuration script which then processes to select the local OVPN file found within the same directory accordingly.

Happy to push this to master and maintain it once I get things fully functional.

**Comment by pkishino** [comment]
@edgd1er perhaps you can help him out here based on your nordvpn stuff.

**Reply by derekcentrico** [reply]
Sure, so I created https://github.com/derekcentrico/docker-transmission-openvpn/ and renamed fetch-external-configs.sh and added in the ovpn directory.  Oddly enough, it still doesn't see the ovpn's directory and sh file.  I'm trying to find a location where the original internal ones may be hard defined, but alas I am not finding anything.  

> Using OpenVPN provider: OVPN
> Running with VPN_CONFIG_SOURCE auto
> No bundled config script found for OVPN. Defaulting to external config
> /etc/openvpn/start.sh: line 87: ./etc/openvpn/fetch-external-configs.sh: No such file or directory
> No VPN configuration provided. Using default.

**Reply by derekcentrico** [reply]
In the event someone is curious, the docker image does contain /etc/openvpn/ovpn with the new contents.

I'm not finding a script that would declare a provider internal vs. external.  It seems to me it is based on if the directory is or is not present.  So, I'm at a loss on moving forward.  Would love to contribute the OVPN setup to the project.

**Reply by derekcentrico** [reply]
> In the event someone is curious, the docker image does contain /etc/openvpn/ovpn with the new contents.
> 
> I'm not finding a script that would declare a provider internal vs. external. It seems to me it is based on if the directory is or is not present. So, I'm at a loss on moving forward. Would love to contribute the OVPN setup to the project.

I'm a fool.  I didn't 755 the script.  Sweet that solves that.

**Comment by derekcentrico** [comment]
Well everything works.  This was a fun learning experience when time permitted.  https://github.com/haugene/docker-transmission-openvpn/pull/2282 should people have a desire to let others benefit from the internal script and variables.

**Reply by ptalanoa** [reply]
man... I am a little lost.  I see the default.ovpn config... if I were to rename that to default.ovpn.old and take the netherlands.ovpn (that I want to use) and rename that to default.ovpn would that work? Would that be persistent between reboots of the docker container? I am running this under Home Assistant via docker and looking at settings via portainer.io....

**Reply by derekcentrico** [reply]
So, the problem you will have right now is that my contribution for contrib is in the main branch but my contribution for the vpn is in the dev branch.  You'll want to switch to dev until they push the changes to main.  @pkishino any ideas on when the next update will be pushed to the main branch?

So for Netherlands, your vars end up like this:

OVPN_PROTOCOL=udp
OVPN_COUNTRY=nl
OVPN_CITY=amsterdam
OVPN_CONNECTION=standard

See https://haugene.github.io/docker-transmission-openvpn/provider-specific/ (formatting issues need fixed).

**Reply by derekcentrico** [reply]
FYI - I just submitted pull requests fixing typos.
